log_parser/log_sequence_matcher.py

```diff
@@ -0,0 +1,61 @@
+from datetime import datetime
+
+from log_parser.event_pattern import EventPattern
+from log_parser.detected_event import DetectedEvent
+from log_parser.log import Log
+
+
+class LogSequenceMatcher:
+    """
+    OngoingLogSequence is responsible for managing the state of a sequence of
+    log entries that match a specific event pattern over time.
+    The sequence is considered expired if the time since its creation exceeds
+    the maximum duration defined in the event pattern.
+    """
+
+    def __init__(self, event_pattern: EventPattern, start_time: datetime):
+        self._event_pattern = event_pattern
+        self._start_time = start_time
+        self._matched_logs = []
+
+    def process_log(self, log: Log) -> bool:
```

Comment:
Maybe `match_log` / `match_next_log`?

That way:

1. Relation to the class name
2. It is more clear that we try to match the log and the return value will be more intuitive

Change documentation accordingly